Elemental/Components/AeTable/AePaginatedTable.razor

```diff
@@ -86,18 +86,26 @@
 
     protected override async Task OnParametersSetAsync()
     {
-        LoadPageItems(true);
         Total = Rows.Count();
+        int PageCount = (int)Math.Ceiling(Total / (decimal)Math.Max(1, PageSize));
+        Page = (Page > PageCount - 1) ? PageCount - 1 : Page;
+        LoadPageItems(true);
     }
 
     protected override RenderFragment DefaultHeader(string header, int index)
     {
         return
-    @<AeTypography class="table-cell" @onclick="() => ToggleSortOnColumn(index)">
+    @<AeTypography class="table-cell" @onclick="() => SortTable(index)">
         @header
         <span class="ae table-sort"><i class="@GetSortIconClass(index)"></i></span>
     </AeTypography>
     ;
     }
 
+    private void SortTable(int index)
+    {
+        ToggleSortOnColumn(index);
+        LoadPageItems(true);
+    }
```

Comment:
Nice, can we rename to something more descriptive? `SortAndRecalculatePageItems`?